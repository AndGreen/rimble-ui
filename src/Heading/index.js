import React from 'react'

import styled from 'styled-components'
import Text from '../Text'

// body size + 6 headings
const Heading = styled(Text)``

Heading.defaultProps = {
  as: 'h3',
  regular: true,
  fontSize: 4,
  my: 3,
}

Heading.h1 = styled(Heading)``
Heading.h1.defaultProps = {
  as: 'h1',
  fontSize: 6,
  my: 3,
}

Heading.h2 = styled(Heading)``
Heading.h2.defaultProps = {
  as: 'h2',
  fontSize: 5,
  my: 3,
}
Heading.h3 = styled(Heading)``
Heading.h3.defaultProps = {
  as: 'h3',
  fontSize: 4,
  my: 3,
}
Heading.h4 = styled(Heading)``
Heading.h4.defaultProps = {
  as: 'h4',
  fontSize: 3,
  my: 3,
}
Heading.h5 = styled(Heading)``
Heading.h5.defaultProps = {
  as: 'h5',
  fontSize: 2,
  my: 3,
}
Heading.h6 = styled(Heading)``
Heading.h6.defaultProps = {
  as: 'h6',
  fontSize: 1,
  my: 3,
}

export default Heading
