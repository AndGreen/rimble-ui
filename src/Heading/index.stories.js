import React from 'react'

import { storiesOf } from '@storybook/react'

import Heading from '../Heading'

storiesOf('Heading', module)
  .add('default', () => (
    <Heading>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading>
  )).add('with h1', () => (
    <Heading.h1>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading.h1>
  )).add('with h2', () => (
    <Heading.h2>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading.h2>
  )).add('with h3', () => (
    <Heading.h3>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading.h3>
  ))
  .add('with h4', () => (
    <Heading.h4>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading.h4>
  ))
  .add('with h5', () => (
    <Heading.h5>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading.h5>
  ))
  .add('with h6', () => (
    <Heading.h6>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam delectus unde porro doloribus, accusantium quasi illum aperiam architecto adipisci voluptatibus illo velit modi expedita ipsa perspiciatis sed iste sit repellat, alias. Soluta sed, cumque, fugit est odit ipsam inventore porro officia repellendus voluptatem neque error. Mollitia quam cupiditate doloribus aliquid. Dolore praesentium, rerum, consequatur tenetur sapiente aliquid distinctio expedita repudiandae dolor quas maxime. Molestiae aspernatur ea beatae velit ratione neque, commodi blanditiis voluptate corporis ex cum dolores eaque animi eveniet odit assumenda deleniti unde corrupti facere veniam quibusdam. Ex at repudiandae iure quae, natus illum eveniet eaque, autem rem dicta, aliquid! Nobis voluptates, ut quae soluta laborum eius cumque minima perferendis ab suscipit. Accusamus atque optio amet esse assumenda, illum temporibus sit veniam minima qui est explicabo error placeat consectetur, ea corporis dignissimos aliquam tempore nobis nemo impedit illo recusandae sint. Eligendi laboriosam, neque fuga, debitis, quasi et corporis ipsam nihil tempora perspiciatis, impedit at! Quam atque nemo maiores rem repellat quae, aliquid, tempora sed ipsam adipisci tenetur explicabo, deleniti debitis. Iure sequi, illo, cum sit obcaecati nostrum molestias aspernatur odio aliquam adipisci esse cupiditate quis nulla nobis impedit qui ab magni natus voluptas saepe et! Ipsum praesentium in tempora.
    </Heading.h6>
  ))
