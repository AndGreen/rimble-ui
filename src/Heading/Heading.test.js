import Heading from './'

describe('Heading component sanity', () => {
  it('has name', () => {
    expect(Heading.displayName).toBe('Heading')
  })

  it('matches default snapshot', () => {
    const component = render(<Heading />)
    expect(component).toMatchSnapshot()
  })
})
