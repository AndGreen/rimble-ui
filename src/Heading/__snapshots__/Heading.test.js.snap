// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component sanity matches default snapshot 1`] = `
.c0 {
  margin-top: 16px;
  margin-bottom: 16px;
  font-size: 24px;
  text-align: left;
  font-size: 24px;
  font-weight: 3;
  text-align: left;
}

.c0 {
  box-sizing: border-box;
}

<h3
  class="c0"
  font-size="4"
  font-weight="3"
>
  
</h3>
`;
